documentstore_migracao/tools/constructor.py

```diff
@@ -7,22 +7,37 @@
 from fs import path
 from fs.walk import Walker
 
-from documentstore_migracao.utils import files, string, xml
+from documentstore_migracao.utils import files, string, xml, pid_manager
 from documentstore_migracao.export.sps_package import SPS_Package
 from documentstore_migracao import config
 
 logger = logging.getLogger(__name__)
 
 
-def article_xml_constructor(file_xml_path: str, dest_path: str, in_place: bool) -> None:
+def article_xml_constructor(file_xml_path: str, dest_path: str, pid_database_engine, in_place: bool) -> None:
 
     logger.debug("file: %s", file_xml_path)
 
     parsed_xml = xml.loadToXML(file_xml_path)
     xml_sps = SPS_Package(parsed_xml)
 
-    # CONSTROI O SCIELO-id NO XML CONVERTIDO
-    xml_sps.create_scielo_id()
+    pid_v2 = xml_sps.scielo_pid_v2
+
+    # VERIFICA A EXISTÊNCIA DO PID V3 NO XC ATRAVES DO PID V2
+    if not pid_manager.check_pid_v3_by_v2(pid_database_engine, pid_v2):
+
+        # CONSTROI O SCIELO-id NO XML CONVERTIDO
+        xml_sps.create_scielo_id()
+
+        # CRIA O PID V2 E V3 NA BASE DE DADOS DO XC
+        pid_manager.create_pid(pid_database_engine, pid_v2, xml_sps.scielo_pid_v3)
+
+    else:
+
+        # SE CASO EXISTA O PID NO VERSÃO 3 NA BASE DO XC É PRECISO ADICIONAR NO XML
+        pid_v3 = pid_manager.get_pid_v3_by_v2(pid_database_engine, pid_v2)
+
+        xml_sps.scielo_pid_v3(pid_v3)
```

Comment:
@jamilatta curiosamente se o artigo já possuir um PID na base de dados mas não no _XML_ o programa não consegue atualizar o XML e o seguinte erro é lançado:

```
The following exception was raised: ''NoneType' object is not callable'.
```

O erro acontece em `document-store-migracao/documentstore_migracao/tools/constructor.py", line 40, in article_xml_constructor`, onde o programa não consegue escrever na árvore do lxml.

Reply:
@joffilyfe estou verificando e pelo que estou vendo o método scielo_pid_v3(pid_v3) não esta funcionando estou testando....

Reply:
Testando localmente realmente o **scielo_pid_v3**(pid_v3) e **scielo_pid_v2**(pid_v2) não está funcionando.

Veja: 

![Screen Shot 2020-08-17 at 16 06 13](https://user-images.githubusercontent.com/373745/90434398-b76ebb80-e0a3-11ea-91eb-c115af946109.png)

Eu e o @rafaelpezzuto testamos em conjunto e realmente esse método não está funcionando corretamente, olhando o código esse método nunca foi utilizado nessa base de código (código inutilizável), sugiro que criemos um novo issue para ajustar esse método.

Reply:
@jamilatta, `scielo_pid_v3` é uma property. Portanto, se você usar `xml_sps.scielo_pid_v3 = pid_v3` ao invés de `xml_sps.scielo_pid_v3(pid_v3)` deve funcionar.

Reply:
@joffilyfe poderia testar novamente.

Reply:
@patymori sim sim rsrs...